parser.js

```diff
@@ -889,13 +888,29 @@
       } else if (match('\\')) {
         res = parseClassEscape();
         if (!res) {
-          throw SyntaxError('classEscape');
+          bail('classEscape');
         }
 
         return parseUnicodeSurrogatePairEscape(res);
       }
     }
 
+    function bail(message, from, to) {
+      from = from == null ? pos : from;
+      to = to == null ? from : to;
+
+      var contextStart = Math.max(0, from - 10),
+          contextEnd = Math.min(to + 10, str.length);
+
+      // Output a bit of context and a line pointing to where our error is.
+      //
+      // We are assuming that there are no actual newlines in the content as this is a regular expression.
+      var context = '    ' + str.substring(contextStart, contextEnd),
+          pointer = '    ' + new Array(from - contextStart + 1).join(' ') + '^';
```

Comment:
nit: Can you please use `var` for both lines instead of a comma to wrap to the next line?

```js
      var context = '    ' + str.substring(contextStart, contextEnd);
      var pointer = '    ' + new Array(from - contextStart + 1).join(' ') + '^';
```

Also applies to the `contextStart` and `contextEnd` variable above.